### PR TITLE
No need to re-browse after search in new API

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
Because search method already returns a recorset, there is no need to re-browse on it.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You can even write directly in return recordset without making a for.

File: account_invoice_merge/invoice.py [hunk 1]
```diff
@@ -188,9 +188,9 @@ def make_key(br, fields):
             if 'purchase.order' in self.env.registry else False
         for new_invoice_id in invoices_info:
             if so_obj:
-                todo_ids = so_obj.search(
+                todos = so_obj.search(
                     [('invoice_ids', 'in', invoices_info[new_invoice_id])])
-                for org_so in so_obj.browse(todo_ids):
+                for org_so in todos:
                     org_so.write({'invoice_ids': [(4, new_invoice_id)]})
```
Comment (yvaucher) [inline account_invoice_merge/invoice.py]: Following @pedrobaeza comment, you can extract write for loop and do

```
todos.write({'invoice_ids': [(4, new_invoice_id)]})
```

File: account_invoice_merge/invoice.py [hunk 2]
```diff
@@ -200,17 +200,16 @@ def make_key(br, fields):
                             so_line.write(
                                 {'invoice_lines': [(6, 0, invoice_line_ids)]})
             if po_obj:
-                todo_ids = po_obj.search(
+                todos = po_obj.search(
                     [('invoice_ids', 'in', invoices_info[new_invoice_id])])
-                for org_po in po_obj.browse(todo_ids):
+                for org_po in todos:
```
Comment (yvaucher) [inline account_invoice_merge/invoice.py]: Remove for loop and do

```
todos.write({'invoice_ids': [(4, new_invoice_id)]})
```

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza @yvaucher Thanks for review. Code is now fixed.

**Comment by yvaucher** [comment]
@Cedric-Pigeon Thanks for the changes :+1:

**Comment by sbidoul** [comment]
:+1: 

The needs fixing tag can be removed.

**Pull request merged**